MAINT: Update for pandas API change

@drammock when you get a chance can you look into https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=24609&view=logs&j=305851a9-a7bb-55db-0042-7e2b6f48aa1c&t=4d37777d-f36a-53aa-9217-6386d15dddcd&l=3448 :

```
================================== FAILURES ===================================
_____ test_unaggregated_spectrum_to_data_frame[multitaper-complex-False] ______
mne\time_frequency\tests\test_spectrum.py:188: in test_unaggregated_spectrum_to_data_frame
    agg_df = gb.apply(_agg_helper, spectrum._mt_weights, grouping_cols)
C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\pandas\core\groupby\groupby.py:1495: in apply
    warnings.warn(
E   FutureWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Select the columns to operate on after groupby toeither explicitly include or exclude the groupings and silence this warning.
______ test_unaggregated_spectrum_to_data_frame[multitaper-complex-True] ______
mne\time_frequency\tests\test_spectrum.py:188: in test_unaggregated_spectrum_to_data_frame
    agg_df = gb.apply(_agg_helper, spectrum._mt_weights, grouping_cols)
C:\hostedtoolcache\windows\Python\3.10.11\x64\lib\site-packages\pandas\core\groupby\groupby.py:1495: in apply
    warnings.warn(
E   FutureWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Select the columns to operate on after groupby toeither explicitly include or exclude the groupings and silence this warning.
```

I'm adding an ignore in https://github.com/mne-tools/mne-python/pull/11616 just to get things green but a proper fix is needed

--- a/mne/time_frequency/tests/test_spectrum.py
+++ b/mne/time_frequency/tests/test_spectrum.py
@@ -176,8 +176,8 @@ def test_unaggregated_spectrum_to_data_frame(raw, long_format, method, output):
     # sorting at the agg step *sigh*
     _inplace(orig_df, 'sort_values', by=grouping_cols, ignore_index=True)
     # aggregate
-    gb = df.drop(columns=drop_cols).groupby(
-        grouping_cols, as_index=False, observed=False)
+    df = df.drop(columns=drop_cols)
+    gb = df.groupby(grouping_cols, as_index=False, observed=False)
     if method == 'welch':
         if output == 'complex':
             def _fun(x):
@@ -186,6 +186,7 @@ def test_unaggregated_spectrum_to_data_frame(raw, long_format, method, output):
             _fun = np.nanmean
         agg_df = gb.aggregate(_fun)
     else:
+        gb = gb[df.columns]  # https://github.com/pandas-dev/pandas/pull/52477
         agg_df = gb.apply(_agg_helper, spectrum._mt_weights, grouping_cols)
     # even with check_categorical=False, we know that the *data* matches;
     # what may differ is the order of the "levels" in the *metadata* for the

--- a/mne/time_frequency/tests/test_spectrum.py
+++ b/mne/time_frequency/tests/test_spectrum.py
@@ -137,8 +137,6 @@ def _agg_helper(df, weights, group_cols):
     return Series(_df)
 
 
-# TODO: Fix this warning
-@pytest.mark.filterwarnings("ignore:.*columns to operate on.*:FutureWarning")
 @pytest.mark.parametrize('long_format', (False, True))
 @pytest.mark.parametrize('method, output', [
     ('welch', 'complex'),

